EIPS/eip-7702.md

```diff
@@ -79,12 +79,18 @@ If any of the above steps fail, immediately stop processing that tuple and conti
 
 Note that the signer of an authorization tuple may be different than `tx.origin` of the transaction.
 
+If transaction execution results in failure (any exceptional condition or code reverting), setting delegation designations is *not* rolled back.
+
 ##### Delegation Designation
 
-The delegation designation uses the banned opcode `0xef` from [EIP-3541](./eip-3541.md) to designate the code has a special purpose. This designator requires all code retrieving operations follow the address pointer to fill the accounts observable code. The following instructions are impacted: `EXTCODESIZE`, `EXTCODECOPY`, `EXTCODEHASH`, `CALL`, `CALLCODE`, `STATICCALL`, `DELEGATECALL`.
+The delegation designation uses the banned opcode `0xef` from [EIP-3541](./eip-3541.md) to designate the code has a special purpose. This designator requires all code retrieving operations follow the address pointer to fill the accounts observable code. The following instructions are impacted: `EXTCODESIZE`, `EXTCODECOPY`, `EXTCODEHASH`, `CALL`, `CALLCODE`, `STATICCALL`, `DELEGATECALL`, as well as transactions with `destination` targeting the code with delegation designation. 
 
 For example, `EXTCODESIZE` would return the size of the code pointed to by `address` instead of `23` which would represent the delegation designation. `CALL` would similarly load the code from `address` and execute it in the context of `authority`.
 
+In case a delegation designator points to a precompile address, retrieved code is considered empty and `CALL`, `CALLCODE`, `STATICCALL`, `DELEGATECALL` instructions targeting this account will execute empty code, i.e. succeed with no execution given enough gas.
```

Comment:
Personally would like to change this rule to the opposite, i.e. executing precompiles, but this is what tests currently expect.

Reply:
What would be the argument for doing this? IMO that conflates the trie structure and it's values with the convenience feature of the EVM of having "precompiles" at specific addresses.

Reply:
Argument would be being consistent in how if execution is targeting certain address, it behaves the same regardless if it was 7702-delegated.

I.e. currently there is a special case "If execution targets a precompile, don't execute it if target was delegated to".

Reply:
Hmm I just don't see 7702 operating at that level. To me it's more about reading the trie and setting certain values in the trie. It's then, separately, the job of the EVM to execute the code it's provided.

Reply:
I guess it depends on where the boundary between EVM and the rest of the client lies in the particular implementation. If EVM is in charge of resolving precompile addresses, it's natural to do the same resolution for 7702 case. If precompile resolution is outside of EVM and EVM gets only code, then it could be more natural to give it empty code in this case.